lib/request.js

```diff
@@ -252,6 +252,7 @@ module.exports = {
   get host() {
     const proxy = this.app.proxy;
     let host = proxy && this.get('X-Forwarded-Host');
+    if (this.req.httpVersionMajor >= 2) host = this.get(':authority');
```

Comment:
`X-Forwarded-Host` will force overwrite at http2?

Reply:
The way I understand the definition of `:authority` is that it can be viewed as a replacement of the `Host` header.
Therefore a Proxy would set the `:authority` header to the backend hostname und set the `X-Forwarded-Host` to the original host.
So the current order seems to be correct.

Reply:
I think we wanna keep the forwarded header, don’t we?

Reply:
If `app.proxy` was set, I'd like to keep the `X-Forwarded-Host` set by reverse proxy(nginx).

 And in my practice, when client request nginx with http 1/2 and nginx proxy to node server, it always use `X-Forwarded-Host`, it is a de-facto standard for reverse proxy to set host when visit upstream.

Reply:
If `app.proxy` was set, I'd like to keep the `X-Forwarded-Host` set by reverse proxy(nginx).

 And in my practice, when client request nginx with http 1/2 and nginx proxy to node server, it always use `X-Forwarded-Host`, it is a de-facto standard for reverse proxy to set host when visit upstream.

Reply:
If `app.proxy` was set, I'd like to keep the `X-Forwarded-Host` set by reverse proxy(nginx).

 And in my practice, when client request nginx with http 1/2 and nginx proxy to node server, it always use `X-Forwarded-Host`, it is a de-facto standard for reverse proxy to set host when visit upstream.

Reply:
If `app.proxy` was set, I'd like to keep the `X-Forwarded-Host` set by reverse proxy(nginx).

 And in my practice, when client request nginx with http 1/2 and nginx proxy to node server, it always use `X-Forwarded-Host`, it is a de-facto standard for reverse proxy to set host when visit upstream.

Reply:
If `app.proxy` was set, I'd like to keep the `X-Forwarded-Host` set by reverse proxy(nginx).

 And in my practice, when client request nginx with http 1/2 and nginx proxy to node server, it always use `X-Forwarded-Host`, it is a de-facto standard for reverse proxy to set host when visit upstream.

Reply:
Ok, so no changes are required.